ext/oci8/oci8_interface.c

```diff
@@ -44,6 +44,29 @@
 #define OCI_STMT_CALL 10
 #endif
 
+/* {{{ proto bool oci_register_taf_callback( resource connection [, string callback] )
+   Register a callback function for Oracle Transparent Application Failover (TAF) */
+PHP_FUNCTION(oci_register_taf_callback)
+{
+	zval *z_connection;
+	php_oci_connection *connection;
+	char *callback = NULL;
+	int callback_len = 0;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r|s!", &z_connection, &callback, &callback_len) == FAILURE) {
```

Comment:
Currently this restricts the callback to being a string. It would be preferable to accept a general "callable" here (such as a closure). For this the `f` modifier is used together with `zend_fcall_info *` and `zend_fcall_info_cache *` arguments. A look at https://github.com/php/php-src/blob/edfa87ebf5bd779c9c913cb509c5e24e35e1177f/ext/standard/array.c#L5997 might help, though it uses a different parameter parsing API.